codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/RestJson1.kt

```diff
@@ -177,40 +188,45 @@ class RestJson1HttpSerializerGenerator(
                     val errorTrait = variantShape.expectTrait<ErrorTrait>()
                     val variantSymbol = symbolProvider.toSymbol(variantShape)
                     val data = safeName("var")
-                    val serializerSymbol =
-                        serde.member(symbolProvider.serializeFunctionName(variantShape))
-                    rustBlock("#TKind::${variantSymbol.name}($data) =>", errorSymbol) {
-                        rust(
-                            """
-                                #T(&mut object, &$data);
-                                object.key(${"code".dq()}).string(${httpBindingResolver.errorCode(variantShape).dq()});
-                            """.trimIndent(),
-                            serializerSymbol
-                        )
-                        if (variantShape.errorMessageMember() != null) {
+                    val serializerSymbol = jsonSerializerGenerator.serverStructureSerializer(variantShape, variantShape.members().toList())
```

Comment:
I think you _probably_ need to use the http index here as well, or at least add a comment that you should.

IIRC, errors can also have members that are set as headers instead of in the body

Reply:
Fixed in next commit.